src/track-promise.cc

```diff
@@ -0,0 +1,72 @@
+#include "env.h"
+#include "env-inl.h"
```

Comment:
It seems you have a copy of the `TrackPromise` class duplicated in this file below – that's why it works.

If nothing else, other than this files, needs to know about the `TrackPromise` class, you don't need `track-promise.h`.